### ISSUE TITLE
Support \donttest{}

### ISSUE THREAD

**Issue by lorenzwalthert** [r-lib/styler#498]
``` r
style_text("
#' Remove the local CITES database
#'
#' @examples
#' \\donttest{
#' \\dontrun{
#' cites_db_delete()
#' }
#' }
cites_db_delete <- function() {
  for (t in dbListTables(cites_db())) {
    dbRemoveTable(cites_db(), t)
  }
  update_cites_pane()
}"
)
#> Error in if (from > to) {: missing value where TRUE/FALSE needed
```

<sup>Created on 2019-04-26 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by lorenzwalthert** [comment]
Seems to be seen as valid code by roygen ans R CMD CHECK, hence we should support it. Reference: https://stackoverflow.com/questions/1454211/what-does-not-run-mean-in-r-help-pages

**Comment by ycphs** [comment]
I had the same problem. The return was meaningless:

```Warning message:
When processing R/style_wrappers.R: <text>:15:1: unexpected input
14: 
15: \

```
If an error occurs it shouldn't be generic return.

I would prefer to get the following information:

- location of the error 
- is it part of the actual code
- is it part of the roxygen structure
- prompt continue to the end of the file with skipping the relevant part of the code (y/n)

**Comment by lorenzwalthert** [comment]
I don't think we are going to support styling this soon. It's pretty complicated to solve this in an elegant way and cover all cases I think. I agree we should fail more informatively. Before parsing roxygen, we could check if there are nested statements and if so, print the code (because exact line information will be harder) in a warning and proceed, so people can search their code for the pattern.

**Comment by IndrajeetPatil** [comment]
Since `CRAN` implemented a new check for [excessive imports](https://github.com/wch/r-source/commit/0639b2c2f152834ac30bb5ed321e912706e34239), I have been seeing a new trend among developers where they conditionally use the needed packages and add them to `Suggests`, and not `Imports`. 
This means there are increasing number of instances where people are using either nested `\donttest \dontrun` or something along those lines. 

So, yes, the need for this feature is increasing, and I am looking forward to this feature eventually being supported by `styler` :)

**Comment by lorenzwalthert** [comment]
@IndrajeetPatil any interest in spending some time to quantify how frequent this is? I have no feeling for this. E.g. what share of packages do use nested `dont`s? Of which form are they (seee above), e.g. all R expression nested with same depth or not?

**Comment by IndrajeetPatil** [comment]
@lorenzwalthert Sure! Will do some digging on a random sample and report back the results.

**Comment by yjunechoe** [comment]
Hi! Just wanted to bump this with a similar issue for nesting `\dontshow{}` in `\donttest{}`. This is common when examples have setup code you want to hide (API auth, establishing connection to a background python/julia/etc. session) that also take too long (beyond time allotted for examples by CRAN).

Examples in a [package](https://github.com/yjunechoe/jlmerclusterperm/) of mine are along the lines of:

```
#' @examples
#' \donttest{
#' \dontshow{
#' # Setup code (ex: establishing connection)
#' }
#' # Actual example
#' \dontshow{
#' # Cleanup code (ex: closing connection)
#' }
#' }
```

I understand this is non trivial to handle, but if there are any workarounds that I can use in the meantime I'd really appreciate it!

**Comment by lorenzwalthert** [comment]
Well, the workaround is to turn styler off for these lines I think.

**Comment by rempsyc** [comment]
My use case is that CRAN requires me to demonstrate the file saving feature in a temporary directory, but that will not be useful to the user. But even turning styler off doesn't seem to work in my case? Am I using it wrong @lorenzwalthert ? Reprex:

``` r
# inline
styler::style_text("
#' @examples
#' \\donttest{
#' \\dontshow{1+1} # styler: off
#' }"
)
#> Error in `map()`:
#> ℹ In index: 1.
#> Caused by error in `purrr::map()`:
#> ℹ In index: 1.
#> Caused by error in `map()`:
#> ℹ In index: 2.
#> Caused by error in `map2()`:
#> ℹ In index: 1.
#> Caused by error in `if (from > to) ...`:
#> ! missing value where TRUE/FALSE needed

# block
styler::style_text("
#' @examples
#' \\donttest{
#' # styler: off
#' \\dontshow{
#'   1+1
#' }
#' # styler: on
#' }"
)
#> Error in `map()`:
#> ℹ In index: 1.
#> Caused by error in `purrr::map()`:
#> ℹ In index: 1.
#> Caused by error in `map()`:
#> ℹ In index: 2.
#> Caused by error in `map2()`:
#> ℹ In index: 1.
#> Caused by error in `if (from > to) ...`:
#> ! missing value where TRUE/FALSE needed

# block variant 2
styler::style_text("
#' # styler: off
#' @examples
#' \\donttest{
#' \\dontshow{1+1}
#' }
#' # styler: on"
)
#> Error in `map()`:
#> ℹ In index: 1.
#> Caused by error in `purrr::map()`:
#> ℹ In index: 1.
#> Caused by error in `map()`:
#> ℹ In index: 2.
#> Caused by error in `map2()`:
#> ℹ In index: 1.
#> Caused by error in `if (from > to) ...`:
#> ! missing value where TRUE/FALSE needed
```

<sup>Created on 2025-09-26 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>
